### ISSUE TITLE
Integration of XProcSpec testing

### ISSUE THREAD

**Issue by rdeltour** [daisy/pipeline-tasks#7]
XProcSpec testing s/b enabled by default in a module's build, via the `xproc-maven-plugin`
### Notes from the Paris f2f
- xprocspec
  - feature requests
    - [x] [p:library](https://github.com/daisy-consortium/xprocspec/issues/5)
      - would help work around issue with circular imports (check if
        this issue can be fixed in fileset-utils)
    - [x] [inline script](https://github.com/daisy-consortium/xprocspec/issues/6)
      - not essential, w/b nice
    - [x] [xfail](https://github.com/daisy-consortium/xprocspec/issues/7)
      - not essential, can use pending for now
    - [x] [error messages](https://github.com/daisy-consortium/xprocspec/issues/11)
      - mostly calabash issue: v1.0.9: no descriptive error messages
        (see ndw/xmlcalabash1#97)
    - [x] [documentation element](https://github.com/daisy-consortium/xprocspec/issues/10)
      - not essential
      - include in html-report?
    - [ ] detailed diffs similar to XSpec
      - [x] for type=compare
      - [ ] for type=custom
  - memory issue
    - e.g. in dtbook-to-odt
    - [x] try Calabash 1.0.18: same problem
    - [x] try with xproc-maven-plugin & custom catalog file: same problem
- xproc-maven-plugin
  - [x] mocking dependencies (support for catalog.xml)
  - [x] html-report more useful then junit-report
    - include both
    - add overview page like in xspec-maven-plugin
  - [x] more logging
    - from XProcSpec
    - from Calabash
  - [x] support for calabash.xml config file

**Comment by josteinaj** [comment]
All the linked xprocspec issues should be fixed now:
- [x] daisy/xprocspec#5
- [x] daisy/xprocspec#6
- [x] daisy/xprocspec#7
- [x] daisy/xprocspec#11
- [x] daisy/xprocspec#10

There's no issue for the "custom comparator could return more detailed result". I'm not sure exactly what that entails so please create an issue for it which describes it in more detail.

I suppose the issues "html-report" (daisy/xproc-maven-plugin#3) and "more logging" (daisy/xproc-maven-plugin#5) are non-essential for v1.8. But we need the ability to provide a custom XML catalog (daisy/xproc-maven-plugin#2) before enabling xprocspec tests in all modules by default, because otherwise we can't perform unit tests. I've created a separate issue for integration tests, but that's probably a post-1.8 feature (daisy/xproc-maven-plugin#7).

What remains after all this is ready? Do we add xproc-maven-plugin to the modules-parent pom? Does this depend on #8 being finished first?

**Comment by bertfrees** [comment]
I'm on daisy/xproc-maven-plugin#2, daisy/xproc-maven-plugin#3 and daisy/xproc-maven-plugin#5. I'm starting with that ASAP. I don't think we depend on #8.

**Comment by josteinaj** [comment]
@bertfrees: great! :)

**Comment by bertfrees** [comment]
The memory issue is still not solved by the way. The issue is there both when using Pax-Exam and when using xproc-maven-plugin with the catalog.xml feature.
